tests/jerry/func-decl.js

```diff
@@ -23,3 +23,27 @@ assert ((function() {
     }
     return f();
 })() === 'bar');
+
+function check_syntax_error (s) {
+  try {
+    eval (s);
+    assert (false);
+  }
+  catch (e) {
+    assert (e instanceof SyntaxError);
+  }
+}
+
```

Comment:
Actually, there is already a test case in the patch that satisfies to described conditions - the function below (lines 37-49).
```
+function test_strict_mode_propagation_in_func_expr_and_getters_setters () {
+  var p = function () {
+    'use strict';
+
+    return true;
+  }
+
+  var o = { get prop () { 'use strict'; return true; }, set prop (v) { 'use strict'; } };
+
+  function test () {
+    eval = 1;
+  }
+}
```